### PR TITLE
🔥 Remove technical details relative to identifiers

### LINKED ISSUE #1048: Remove id from model classes
## Feature Request

The id should not be part of the model classes. The outside world shouldn't know anything about the technical details. Depending on the user application, the id could be a string, uuid or an integer.

Classes:
- `Model\Page(Interface)`
- `Model\Site(Interface)`

### PR DESCRIPTION
**Pull request by Th3Mouk**
## Subject

I am targeting this branch, because changes are BC break.

Closes #1048 

## Changelog

```markdown
### Removed
- Remove getters and setters relative to identifiers in interfaces and abstract classes
```

### CODE REVIEW

**Comment by core23** [comment]
Please have a look at the failing tests @Th3Mouk

**Comment by Th3Mouk** [comment]
@core23 test are passing locally with `phpunit -c phpunit.xml.dist` on `7.5.16` any idea why ?

**Comment by OskarStark** [comment]
Looks like you cannot mock getId() on Pageinterface anymore, as you removed it

**Comment by OskarStark** [comment]
Any plans to work on this again @Th3Mouk ?

**Comment by Th3Mouk** [comment]
Still need help, tests passing locally, so it's long to debug it

**Comment by OskarStark** [comment]
Can you please do a rebase? Maybe that helps?

File: src/Model/PageInterface.php [hunk 1]
```diff
@@ -24,15 +24,6 @@ interface PageInterface
 {
     public const PAGE_ROUTE_CMS_NAME = 'page_slug';
 
-    /**
-     * Returns the id.
-     *
-     * @return mixed
-     */
-    public function getId();
```
Comment (wbloszyk) [inline src/Model/PageInterface.php:32]: IMHO this method should be keep in interfaces

File: src/Model/SiteInterface.php [hunk 2]
```diff
@@ -25,8 +25,6 @@ interface SiteInterface
      */
     public function __toString();
 
-    public function getId();
```
Comment (wbloszyk) [inline src/Model/SiteInterface.php:28]: The same here. Only setter should be delete from interfaces.

**Pull request closed without merging**